app/src/main/assets/debug/DebugActivity.java

```diff
@@ -1,76 +1,63 @@
-
-                        madeErrorMessage += split[0].substring(addIndex, split[0].length());
-                        madeErrorMessage += "\n\nDetailed error message:\n" + errorMessage;
-                        break;
-                    }
-                }
-
-                if (madeErrorMessage.isEmpty()) {
-                    madeErrorMessage = errorMessage;
-                }
-            } catch (Exception e) {
-                madeErrorMessage = madeErrorMessage + "\n\nError while getting error: " + Log.getStackTraceString(e);
-            }
-        }
-
-        AlertDialog dialog = new AlertDialog.Builder(this)
-                .setTitle("An error occurred")
-                .setMessage(madeErrorMessage)
-                .setPositiveButton("End Application", new DialogInterface.OnClickListener() {
+        dialog = new AlertDialog.Builder(this)
+                .setTitle("Crash Log")
+                .setMessage(getLastSavedErrorLog(this))
+                .setNeutralButton("Dismiss", null)
+                .setPositiveButton("Dismiss & Clear", new DialogInterface.OnClickListener() {
                     @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        finish();
+                    public void onClick(DialogInterface dialogInterface, int which) {
+                        clearErrorLog(DebugActivity.this);
                     }
                 })
                 .create();
+
+        dialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
+            @Override
+            public void onDismiss(DialogInterface dialogInterface) {
+                DebugActivity.this.finish();
+            }
+        });
+        dialog.setOnShowListener(new DialogInterface.OnShowListener() {
+            @Override
+            public void onShow(DialogInterface dialogInterface) {
+                ((TextView) dialog.findViewById(android.R.id.message)).setTextIsSelectable(true);
+            }
+        });
         dialog.show();
-        ((TextView) dialog.findViewById(android.R.id.message)).setTextIsSelectable(true);
     }
-}
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        if (dialog != null && dialog.isShowing())
+            dialog.dismiss();
+    }
+
+    @SuppressLint("ApplySharedPref")
+    public static void saveErrorLog(Context context, String errorLog) {
+        context.getSharedPreferences("SK_CRASH_LOG", Context.MODE_PRIVATE).edit().putString("errorLog", errorLog).commit();
+    }
+
+    public static void clearErrorLog(Context context) {
+        context.getSharedPreferences("SK_CRASH_LOG", Context.MODE_PRIVATE).edit().remove("errorLog").apply();
+    }
+
+    public static String getLastSavedErrorLog(Context context) {
+        return context.getSharedPreferences("SK_CRASH_LOG", Context.MODE_PRIVATE).getString("errorLog", "");
+    }
```

Comment:
I'm not sure if these should be added. Also, using the application context stored by the default application class might be better.

Reply:
It's generally a better idea and practice to pass the context along. And about methods,
Bro sketchware is using a whole activity for only showing a single TextView, so adding some helper methods instead of putting them in application class doesn't make sense.